engine/equivalence_source_target.ml

```diff
@@ -35,13 +35,26 @@ let constructor_to_domain repr_env : constructor -> domain = function
      | Tag t -> Domain.tag (Domain.Set.point t)
 
 let constrained_subtrees repr_env children fallback =
+  let head_domain kst_list =
+    let _head_domain : constructor -> Domain.t = function
+      | String _ -> failwith "Not implemented"
```

Comment:
This is perfectly fine but actually we know the domain here, it is `Domain.tag Obj.string_tag`. (`Obj` is a module from the OCaml standard library that exports internal/runtime information, `string_tag` is the tag of all string values.)